docs/upgrade-guide.md

```diff
@@ -41,6 +46,34 @@ Upgrading the cluster in this manner keeps the cluster online and able to servic
 - The cluster is less fault tolerant while a node is offline for an upgrade because the cluster requires a quorum of nodes to be online to service write requests.
 - Replicating large amounts of data to a node can have a performance impact on the Leader in the cluster.
 
+### Breaking changes from version 23.10 and earlier
+
+#### External TCP API removed
+
+The external TCP API has been removed in 24.2.0. This affects external clients using the TCP API and configurations related to it.
+
+A number of configuration options have been removed as part of this. EventStoreDB will not start by default if any of the following options are present in the database configuration:
+- `AdvertiseTcpPortToClientAs`
+- `DisableExternalTcpTls`
+- `EnableExternalTcp`
+- `ExtHostAdvertiseAs`
+- `ExtTcpHeartbeatInterval`
+- `ExtTcpHeartbeatTimeout`
+- `ExtTcpPort`
+- `ExtTcpPortAdvertiseAs`
+- `NodeHeartbeatInterval`
+- `NodeHeartbeatTimeout`
+- `NodeTcpPort`
+- `NodeTcpPortAdvertiseAs`
+
+#### Changes to Filtered $all subscription checkpoint intervals
+
+The behavior for issuing checkpoints in filtered `$all` subscriptions has changed. Checkpoints will now be issued at least once per interval, with adjustments during catch-up and live phases.
+- When live, checkpoints will still be issued after exactly checkpoint interval events.
+- When catching up, checkpoints will still be issued after exactly checkpoint interval events (except that the first checkpoint will be offset by one event).
+- When transitioning from catch-up to live, a checkpoint is issued.
+- When falling behind from live to catch-up, a checkpoint is issued.
+
```

Comment:
```suggestion
```

This is a change in behaviour that could be visible to a client, but I don't think it's worth mentioning in the upgrade guide. The idea of the checkpoint is just to give the client a more up to date place to resubscribe from, and we are still satisfying that. This could be seen as a breaking change if someone was using it to try to count events in the underlying log, but it wouldn't have worked for that before anyway because of epochs and the events that we exclude from the default all filter